### ISSUE TITLE
Non-consistent error handling

### ISSUE THREAD

**Issue by pallix** [alfert/propcheck#28]
When running a test where the `precondition` function is missing the first error is:

``` ** (UndefinedFunctionError) function FsmTrialTest.precondition/4 is undefined or private```

When running `mix test` a second time (no modification to the source) the error becomes:

```
Property Elixir.FsmTrialTest.property counter works properly() failed. Counter-Example is:
     {:error, :non_boolean_result}
```

**Comment by alfert** [comment]
It is perfectly clear, why this happens. The first one is the initial error message. The second one takes the initial error as a counter example and tries to re-execute in a property. This fails, because it is no property at all (which boils down to a boolean value). Hence this irritating error message. This is at least a usability bug.

**Comment by pallix** [comment]
Thanks for the explanation.

**Comment by alfert** [comment]
@evnu fixed this one in #36
